src/pages/signup.tsx

```diff
@@ -0,0 +1,115 @@
+import Link from "next/link";
+import { notFound } from "next/navigation";
+import { useRef } from "react";
+import { useRouter } from "next/router";
+
+export default function Signup() {
+  const emailRef = useRef<HTMLInputElement>(null);
+  const usernameRef = useRef<HTMLInputElement>(null);
+  const passwordRef = useRef<HTMLInputElement>(null);
+  const router = useRouter();
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    const email = emailRef.current?.value || "";
+    const username = usernameRef.current?.value || "";
+    const password = passwordRef.current?.value || "";
+    try {
+      const res = await fetch(
+        `https://bookclub-backend.nn.r.appspot.com/api/v1/register`,
```

Comment:
the call should be sent to nextjs backend before passing it to the other backend